src/System.IO.Compression/src/System/IO/Compression/DeflateManaged/InflaterManaged.cs

```diff
@@ -134,8 +137,20 @@ public int Inflate(byte[] bytes, int offset, int length)
                     _formatReader.Validate();
                 }
             }
+            
+            return RestrictStreamWithUnCompressedSize(count);
```

Comment:
Are you concerned that the byte array will still be filled with inflated data even if it's at or past the limit?  For example: once you hit the limit and call this method again it will still do work then return 0.  Maybe you should restrict the length going in?

Reply:
When it reach the limit within this  RestrictStreamWithUnCompressedSize method we are setting `_state = InflaterState.Done;` and `_output._bytesUsed = 0;` which informs the caller inflation is done so there will be no more call `Inflate(...)` method. Not sure if it answers your point …

Reply:
I see. I think you’ll still fill the buffer past count on the limited call.  Is there a reason you can’t limit before doing the work?

Reply:
As there is no way to predict how much size would produce given compressed bytes after infaltion can't limit before inflate, I will move the check into above do while loop, it will not make copied output exactly equeal to expected size but will be closer than calling this method outside of the loop. Anyways calling routines rely on returned count of bytes i think having little more extra bytes in the buffer will not be an issue

Reply:
Surely you can limit up front since it's already limiting to a buffer size.

Reply:
It seems like right before the do/while:

```C#
if (_uncompressedSize > -1)
{
    if (_currentInflatedCount => _uncompressedSize)
    {
        // If throwing is the right thing to do here then the degenerate case of _uncompressedSize == 0 should be tested
        // Also, I'm assuming that if it was completed successfully already we wouldn't be calling here.  If we are, the throw should be elsewhere, I guess.
        throw new Something():  ?
    }
    length = Math.Min(length, _uncompressedSize - _currentInflatedCount);
}
```

Now it won't over-copy.

Reply:
By my understanding we cannot update length, length is used to check if the buffer is filled (inflated output data size can be bigger than this buffer or vise versa). i have tried with updating length, unexpected output